### ISSUE TITLE
Standardized numeric error codes

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry-sdk#1118]
## The Issue
Numeric error codes were discussed sometime back on Slack and more recently here: https://github.com/lbryio/lbry-app/issues/1025#issuecomment-366457165 

The next steps would be to:
1) use error codes in all new exceptions/errors
2) build a list of all places where error codes can be introduced. 
3) communicate with error codes via documentation
4) communicate changes to app team

Slack messages for context:
```
kauffj [1:56 PM]
And yes, I strongly support beginning to introduce numeric codes for each type of error
we are already string checking error messages in at least one place in the app, which is terribly fragile
kauffj [1:59 PM]
just start, assuming you are throwing unique exceptions for errors, just add some property to them called ErrorCode
this isn't something that needs to be done at all at once to get started, just set the rule that all new exceptions/errors will include a numeric code
```

## System Configuration

 

- LBRY Daemon version:
- LBRY App version:
- LBRY Installation ID:
- Operating system:

**Comment by kauffj** [comment]
@jackrobison @eukreign This one seems very easy to get started; we simply need to determine the signature pattern. It's not necessary to add codes to all errors to complete this ticket, but determining the pattern would at least allow them to be added for all future errors and a few key errors as necessary.

If the design is simple/obvious enough, then I suspect even the app team will PR error codes as necessary.

**Comment by kauffj** [comment]
Plus, then @eukreign could add a test to prevent new exceptions without codes :grin:

**Comment by eukreign** [comment]
Work in progress here: https://github.com/lbryio/lbry/wiki/Error-Codes

**Comment by alyssaoc** [comment]
@eukreign Can this be closed?

**Comment by eukreign** [comment]
No, this needs to be implemented in lbry still.

**Comment by lyoshenka** [comment]
fixed by #2611
